src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -97,6 +98,11 @@ internal partial class Binder
                 return CreateTupleLiteralConversion(syntax, (BoundTupleLiteral)source, conversion, isCast, destination, diagnostics);
             }
 
+            if (conversion.Kind == ConversionKind.ImplicitDefault)
+            {
+                return new BoundDefaultOperator(source.Syntax, destination);
+            }
```

Comment:
Makes sense. Initially, I thought that the `default` itself should be updated to carry the type, but now the conversion carries the type that is inferred.